packages/cli/src/commands/setup/deploy/providers/render.js

```diff
@@ -0,0 +1,121 @@
+import fs from 'fs'
+import path from 'path'
+
+import { getPaths } from 'src/lib'
+
+const PROJECT_NAME = getPaths().base.match(/[^/|\\]+$/)[0]
+
+const RENDER_YAML = (database) => {
+  return `services:
+- type: web
+  name: ${PROJECT_NAME}-web
+  env: static
+  buildCommand: yarn rw deploy render web
+  staticPublishPath: ./web/dist
+  envVars:
+  - key: NODE_VERSION
+    value: 14
+  routes:
+  - type: rewrite
+    source: /.redwood/functions/*
+# replace destination api url after first deploy to Render
+    destination: replace_me_with_api_url/*
+  - type: rewrite
+    source: /*
+    destination: /index.html
+
+- type: web
+  name: ${PROJECT_NAME}-api
+  env: node
+  buildCommand: yarn rw deploy render api
+  startCommand: yarn rw serve api
+  envVars:
+  - key: NODE_VERSION
+    value: 14
+${database}
+`
+}
+
+const POSTGRES_YAML = `  - key: DATABASE_URL
+    fromDatabase:
+      name: ${PROJECT_NAME}-db
+      property: connectionString
+
+databases:
+  - name: ${PROJECT_NAME}-db
+`
+
+const SQLITE_YAML = `  - key: DATABASE_URL
+    value: file:./data/sqlite.db
+  disk:
+    name: sqlite-data
+    mountPath: /opt/render/project/src/api/db/data
+    sizeGB: 1`
+
+const RENDER_HEALTH_CHECK = `// render-health-check
+export const handler = async () => {
+  return {
+    statusCode: 200,
+  }
+}
+`
+// prisma data source check
+export const prismaDataSourceCheck = (database) => {
+  if (database === 'none') {
+    return {
+      path: path.join(getPaths().base, 'render.yaml'),
+      content: RENDER_YAML(''),
+    }
+  }
+  const content = fs.readFileSync(getPaths().api.dbSchema).toString()
+  const detectedDatabase = content.match(
+    /(?<=datasource DS.*\n\W*provider\W*)\w+/
```

Comment:
Ah, barely caught this but for having the following artifact in my schema.prisma:
```
datasource DS {
  // example: provider = "sqlite" | "postgresql"
  provider = "postgresql"
  url      = env("DATABASE_URL")
}
```

Two things here:
1. The "DS" is a variable and we'll be renaming to the new Prisma boilerplate, which is "DB". 
2. There may be one or more lines of comments between "datasource" line and "provider"

So the regex is too strict. And the commented lines make for a false match.

@peterp is there any way we can get the datasource provider using Prisma SDK? I did a quick check but couldn't find anything obvious ('cause no docs for them!!) Any other suggestions here?

Reply:
^^ @dac09 and @peterp

Reply:
@SEANDOUGHTY check out this beautiful helper from the Prisma team 🚀
```
import {
  getSchema,
  getConfig,
} from '@prisma/sdk'
```
```
const schema = await getSchema(args['--schema'])
const config = await getConfig({
  datamodel: schema,
})
// here!
config.datasources[0].provider
```

And we're already importing prisma sdk in the CLI package here:
https://github.com/redwoodjs/redwood/blob/8f874c5ad6d4065588f680d476a082951c953faa/packages/cli/package.json#L16

Reply:
OMG. What? They even have an open issue for this.